### PR TITLE
finished assessment

### PR DESCRIPTION
**Pull request by arnoldc94**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [ ✅ ]Coding Challenges
- [ ✅ ] Coding Concepts
- [ ✅ ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:
I would like to know my ranking if it is possible.

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -14,56 +14,202 @@
...
+    expect(divisibleByThree(num1)).toEqual(`${num1} is divisible by three`)
+    // Expected output: "15 is divisible by three"
+    
+    expect(divisibleByThree(num2)).toEqual(`${num2} is divisible by three`)
+    // Expected output: "0 is divisible by three"
+
+    expect(divisibleByThree(num3)).toEqual(`${num3} is not divisible by three`)
+    // Expected output: "-7 is not divisible by three"
+
+     expect(divisibleByThree("Four")).toEqual(`Four is not a number`)
```
Comment (elysemontano) [inline code-challenges.test.js:77]: Awesome!  Great job in testing edge cases!!

File: code-challenges.test.js [hunk 2]
```diff
@@ -14,56 +14,202 @@
...
+
+const divisibleByThree = (number) => {
+  if(typeof number !== "number"){
+    return `${number} is not a number`
+  }else if(number % 3 !== 0){
+    return `${number} is not divisible by three`
+  }else{
+    return `${number} is divisible by three`
+  }
+}
```
Comment (elysemontano) [inline code-challenges.test.js:106]: Excellent work!  Pseudocode is perfect, and code looks great!

File: code-challenges.test.js [hunk 3]
```diff
@@ -14,56 +14,202 @@
...
+// if the statement returns false use an else statement to return `${str} is not a word`
+// if outside of jest testing log function to see result
+
+ const firstVowel = (str) =>{ 
+    if(typeof str === 'string'){
+      return str.search(/[aeiou]/i)
+    }else{
+      return `${str} is not a word`
+    }
+ }
```
Comment (elysemontano) [inline code-challenges.test.js:206]: Outstanding work!  Great research on using the search method and providing detailed pseudocode that explains wonderfully on what is happening there.  I want to encourage you to share more of these answers when we go through the assessment together in class.

File: coding-concepts.js [hunk 4]
```diff
@@ -67,7 +67,7 @@ class Learn {
   }
 }
 var learnStudent = new Learn("George")
-// console.log(learnStudent)
+ console.log(learnStudent)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer:Learn { student: 'George', cohort: 'Bravo', year: 2022 }
+// b) Verify and explain:{ student: 'George', cohort: 'Bravo', year: 2022 }. This is the result because you are creating a new object with the Learn class. the constructor acts like a blueprint which creates and initializes the new object. The name paramater passes to this.name = name. creating a new name variable for the object. this is a keyword used to bind it to the new object that is constructed(learnStudent). this is the same for the other information in the class constructor but you do not need a paramater because the information is already defined in the class constructor. The new object is then created by being stored in the new objects name and is created using a new operator witch is set to create a new instance of the Learn class and passes in the name paramater "George".
```
Comment (elysemontano) [inline coding-concepts.js:73]: This entire section === 💯!

File: interview-questions.md [hunk 5]
```diff
@@ -8,62 +8,63 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: Hoisting is the act of moving the  declaration of functions, variables and classes higher in their scope befor the code is executed. This can be safely done with functions so that you may use them prior to them being executed in the code. With variables and classes the hoisted declarations are not ran until the code is executed.
 
-  Researched answer:
+  Researched answer: A proceess where the interpreter appears to move declarations of functions, classes and variables to the top of its scope befor the execution of its code.
```
Comment (elysemontano) [inline interview-questions.md:53]: Incredible answers in this section!  Thank you for the hard work you put into this.

**Review by elysemontano — APPROVED** [review]
Corey, this assessment is what I picture as a perfect assessment and is at a Competent skill level.  You provide incredible pseudocode, tests that look great, and code that has obviously been taken to the next level by including more complex methods and edge cases.  Your answers and researched answers for both the code concepts and interview sections are top notch!  I really want to encourage you to volunteer your answers more often when we go through this in class.
